tools/python/booking_hotels.py

```diff
@@ -0,0 +1,199 @@
+            payload = stream.read()
+            print(payload)
+            return json.loads(payload)
+
+        except Exception as e:
+            logging.error('Error: {0} Context: {1}'.format(e, payload))
+            return None
+
+
+def make_record(src, rate):
+    return Hotel(
+        unicode(src['hotel_id']),
+        unicode(src['location']['latitude']),
+        unicode(src['location']['longitude']),
+        unicode(src['name']),
+        unicode(src['address']),
+        unicode(src['class']),
+        unicode(rate),
+        unicode(src['ranking']),
+        unicode(src['review_score']),
+        unicode(src['url'])
+    )
+
+
+def download(user, password, path):
+    '''
+    Downloads all hotels from booking.com and stores them in a bunch of .pkl files.
+    '''
+    api = BookingApi(user, password)
+
+    maxrows = 1000
+    countries = api.call("getCountries", dict(languagecodes='en'))
+    for country in countries:
+        countrycode = country['countrycode']
+        logging.info(u'Download[{0}]: {1}'.format(countrycode, country['name']))
+
+        allhotels = []
+        while True:
+            hotels = api.call('getHotels',
+                              dict(new_hotel_type=1, offset=len(allhotels), rows=maxrows, countrycodes=countrycode))
+            if isinstance(hotels, dict) and 'ruid' in hotels:
+                logging.error('Api call failed with error: {0} Code: {1}'.format(hotels['message'], hotels['code']))
+                exit(1)
+
+            allhotels.append(hotels)
+
+            if len(hotels) < maxrows:
+                break
+
+        logging.info('Num of hotels: {0}'.format(len(allhotels)))
+        filename = os.path.join(path,
+                                '{0} - {1}.pkl'.format(country['area'].encode('utf8'), country['name'].encode('utf8')))
+        with open(filename, 'wb') as fd:
+            pickle.dump(allhotels, fd, pickle.HIGHEST_PROTOCOL)
+
+
+def translate(source, output):
+    '''
+    Reads *.pkl files and produces a single list of hotels as tab separated values.
+    '''
+    files = [filename for filename in os.listdir(source) if filename.endswith('.pkl')]
+
+    data = []
+    for filename in files:
+        logging.info('Processing {0}'.format(filename))
+        with open(filename, 'rb') as fd:
+            data += pickle.load(fd)
+
+    # Dict of dicts city_id -> { currency -> [prices] }
+    cities = defaultdict(lambda: defaultdict(list))
+
+    def valid(hotel):
+        return 'city_id' in hotel and 'currencycode' in hotel and 'minrate' in hotel and hotel['minrate'] is not None
+
+    # Collect prices
+    for hotel in data:
+        if valid(hotel):
+            cities[hotel['city_id']][hotel['currencycode']].append(float(hotel['minrate']))
+
+    # Replaces list of prices by a median price.
+    for city in cities:
+        for cur in cities[city]:
+            cities[city][cur] = sorted(cities[city][cur])[len(cities[city][cur]) / 2]
```

Comment:
It's a library method, not overkill. And it's much safer and better to use them instead of your own implementations.